plugins/core/reporter/pprof_manager.go

```diff
@@ -0,0 +1,330 @@
+// Licensed to Apache Software Foundation (ASF) under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Apache Software Foundation (ASF) licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package reporter
+
+import (
+	"context"
+	"io"
+	"strconv"
+	"time"
+
+	"github.com/apache/skywalking-go/plugins/core/operator"
+	commonv3 "github.com/apache/skywalking-go/protocols/collect/common/v3"
+	pprofv10 "github.com/apache/skywalking-go/protocols/collect/pprof/v10"
+)
+
+const (
+	// max chunk size for pprof data
+	maxChunkSize = 1 * 1024 * 1024
+	// max send queue size for pprof data
+	maxPprofSendQueueSize = 30000
+)
+
+type PprofTaskCommand interface {
+	GetTaskID() string
+	GetCreateTime() int64
+	GetDuration() time.Duration
+	StartTask() (io.Writer, error)
+	StopTask(io.Writer)
+	IsDirectSamplingType() bool
+}
+type PprofReporter interface {
+	ReportPprof(taskID string, content []byte)
+}
+
+var NewPprofTaskCommand func(taskID, events string, duration time.Duration,
+	createTime int64, dumpPeriod int, pprofFilePath string,
+	logger operator.LogOperator, manager PprofReporter) PprofTaskCommand
+
+type PprofTaskManager struct {
+	logger         operator.LogOperator
+	serverAddr     string
+	pprofInterval  time.Duration
+	PprofClient    pprofv10.PprofTaskClient // for grpc
+	connManager    *ConnectionManager
+	entity         *Entity
+	pprofFilePath  string
+	LastUpdateTime int64
+	commands       PprofTaskCommand
+	pprofSendCh    chan *pprofv10.PprofData
+}
+
+func NewPprofTaskManager(logger operator.LogOperator, serverAddr string,
+	pprofInterval time.Duration, connManager *ConnectionManager,
+	pprofFilePath string) (*PprofTaskManager, error) {
+	pprofManager := &PprofTaskManager{
+		logger:        logger,
+		serverAddr:    serverAddr,
+		pprofInterval: pprofInterval,
+		connManager:   connManager,
+		pprofFilePath: pprofFilePath,
+		pprofSendCh:   make(chan *pprofv10.PprofData, maxPprofSendQueueSize),
+	}
+	if pprofInterval > 0 {
```

Comment:
When will the interval be less than zero? If less than zero, should we return an error?